### PR TITLE
SEQNG-1157 Add a timeout to observes

### PR DESCRIPTION
**Pull request by cquiroz**
If the instrument observe were to fail (e.g the instrument crashes) the seqexec waits for ever and becomes mostly non-operational
This PR adds a timeout to the observe. This is a bit hard to test but I managed to reproduce it on a simulated GPI and added a test
We should consider adding timeouts to other parts of the obs process

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/ObserveActions.scala [hunk 1]
```diff
@@ -124,7 +127,8 @@ trait ObserveActions {
       _ <- notifyObserveStart(env)
       _ <- env.headers(env.ctx).traverse(_.sendBefore(env.obsId, fileId))
       _ <- info(s"Start ${env.inst.resource.show} observation ${env.obsId.format} with label $fileId")
-      r <- env.inst.observe(env.config)(fileId)
+      t <- env.inst.calcObserveTime(env.config)
+      r <- env.inst.observe(env.config)(fileId).timeout((1.5 * t.toMilliseconds).milliseconds)
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/ObserveActions.scala]: The 1.5 is fairly arbitrary. maybe it should be a constant like 1 min. The idea being that we cannot wait exactly obs time as there are overheads, etc but don't want to wait to long to declare it failed

**Comment by cquiroz** [comment]
I added a few changes:
* Made the timeout shorter, to a flat value of 1 min
* Added timeouts for other observation operations, like stop, resume, etc

**Comment by cquiroz** [comment]
I tested this with the instrument and it successfully handled the timeout after 1m
screencast 2019-11-19 15-34-18

However when I tried to run another time acm produced an exception
```
seqexec_web_server edu.gemini.epics.acm.CaCommandInProgress: null
seqexec_web_server 	at edu.gemini.epics.acm.CaObserveSenderImpl.post(CaObserveSenderImpl.java:211)
seqexec_web_server 	at edu.gemini.epics.acm.CaObserveSenderImpl.postCallback(CaObserveSenderImpl.java:236)
seqexec_web_server 	at seqexec.server.ObserveCommand.$anonfun$post$5(EpicsUtil.scala:126)
seqexec_web_server 	at scala.Option.map(Option.scala:163)
seqexec_web_server 	at seqexec.server.ObserveCommand.$anonfun$post$4(EpicsUtil.scala:124)
seqexec_web_server 	at seqexec.server.ObserveCommand.$anonfun$post$4$adapted(EpicsUtil.scala:123)
seqexec_web_server 	at cats.effect.IO$.$anonfun$async$1(IO.scala:1102)
seqexec_web_server 	at cats.effect.IO$.$anonfun$async$1$adapted(IO.scala:1100)
seqexec_web_server 	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:341)
seqexec_web_server 	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:119)
seqexec_web_server 	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:355)
seqexec_web_server 	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:376)
seqexec_web_server 	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:316)
seqexec_web_server 	at cats.effect.internals.IOShift$Tick.run(IOShift.scala:36)
seqexec_web_server 	at cats.effect.internals.PoolUtils$$anon$2$$anon$3.run(PoolUtils.scala:51)
seqexec_web_server 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
seqexec_web_server 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
seqexec_web_server 	at java.lang.Thread.run(Thread.java:748)
```

File: modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala [hunk 2]
```diff
@@ -143,6 +143,9 @@ trait ObserveCommand {
   def setTimeout[F[_]: Sync](t: Time): F[Unit] =
     Sync[F].delay {
       cs.map(_.getApplySender).map(_.setTimeout(t.toMilliseconds.toLong, MILLISECONDS))
+    } *>
+    Sync[F].delay {
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: @jluhrs I traced how the timeouts on CA are set and I think we are missing this one. It seems this is the core reason for this bug. I'll try to test it on instrument

Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: Yes! This is the core reason. But the new line must replace the old one. This
```
    Sync[F].delay {
      cs.map(_.getApplySender).map(_.setTimeout(t.toMilliseconds.toLong, MILLISECONDS))
    } *>
```
must go away.

**Pull request merged**